score-xml/src/main/java/io/cloudslang/content/xml/actions/EditXml.java

```diff
@@ -0,0 +1,659 @@
+                    break;
+                }
+                case append: {
+                    result.put(Constants.OutputNames.RETURN_RESULT, append(xml, filePath, xpath1, value, type, name));
+                    break;
+                }
+                case subnode: {
+                    result.put(Constants.OutputNames.RETURN_RESULT, subnode(xml, filePath, xpath1, value));
+                    break;
+                }
+                case move: {
+                    result.put(Constants.OutputNames.RETURN_RESULT, move(xml, filePath, xpath1, xpath2));
+                    break;
+                }
+                case rename: {
+                    result.put(Constants.OutputNames.RETURN_RESULT, rename(xml, filePath, xpath1, name, type, value));
+                    break;
+                }
+                case update: {
+                    result.put(Constants.OutputNames.RETURN_RESULT, update(xml, filePath, xpath1, value, type, name));
+                    break;
+                }
+            }
+            result.put(Constants.OutputNames.RETURN_CODE, String.valueOf(Constants.ReturnCodes.SUCCESS));
+        } catch (IllegalArgumentException e) {
+            result.put(Constants.OutputNames.EXCEPTION, "Invalid action " + action);
+            result.put(Constants.OutputNames.RETURN_CODE, String.valueOf(Constants.ReturnCodes.FAILURE));
+        } catch (Exception e) {
+            result.put(Constants.OutputNames.EXCEPTION, e.getMessage());
+            result.put(Constants.OutputNames.RETURN_CODE, String.valueOf(Constants.ReturnCodes.FAILURE));
+        }
+        return result;
+    }
+
+    private ActionsEnum getAction(String action) {
+        return ActionsEnum.valueOf(action.toLowerCase());
+    }
+
+    /**
+     * Validates the operation inputs.
+     *
+     * @param xml      the xml as String
+     * @param filePath the path/remote path to the file
+     * @param action   the edit action
+     * @param xpath1   the first XPath
+     * @param xpath2   the second XPath
+     * @param type     the type on which the edit should occur
+     * @param name     the name of the attribute
+     * @throws Exception for invalid inputs
+     */
+    private void validateInputs(String xml, String filePath, String action, String xpath1, String xpath2, String type, String name, String features) throws Exception {
+        if ((filePath == null || StringUtils.isEmpty(filePath)) && (xml == null || StringUtils.isEmpty(xml))) {
+            throw new Exception("Supplied parameters: either file path or xml is missing when one is required");
+        }
+        if ((!StringUtils.isEmpty(filePath)) && (!StringUtils.isEmpty(xml))) {
+            throw new Exception("Supplied parameters: file path and xml when only one is required");
+        }
+        if (action == null || StringUtils.isEmpty(action)) {
+            throw new Exception("ACTION input is required.");
+        }
+        if (xpath1 == null || StringUtils.isEmpty(xpath1)) {
```

Comment:
@tethryus I guess you can extract whole validations into a generic method ... pass only what you want to verify and the error message